### PR TITLE
cli branch

### PR DESCRIPTION
**Pull request by shiyuhang0**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add branch for ticloud cli

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-cloud/ticloud-branch-create.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+Create a branch:
+
+```shell
+ticloud branch create [flags]
+```
+
+> **Note:**
+>
+> Currently, you can only create branch under [TiDB Serverless](/tidb-cloud/select-cluster-tier.md#tidb-serverless-beta).
```
Comment (zhangyangyu) [inline tidb-cloud/ticloud-branch-create.md:16]: better change a link. beta will be removed I think

File: tidb-cloud/ticloud-branch-delete.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+Delete a branch in interactive mode:
+
+```shell
+ticloud branch delete
+```
+
+Delete a branch in non-interactive mode:
+
+```shell
+ticloud cluster branch --branch-id <branch-id> --cluster-id <cluster-id>
```
Comment (zhangyangyu) [inline tidb-cloud/ticloud-branch-delete.md:31]:
```suggestion
ticloud branch delete --branch-id <branch-id> --cluster-id <cluster-id>
```

File: tidb-cloud/ticloud-branch-list.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+ticloud branch list <cluster-id> -o json
+```
+
+## Args
+
+Branch list command has the following arguments:
+
+| Argument Index | Description                                         | Required | Note                                  |
+|----------------|-----------------------------------------------------|----------|---------------------------------------|
+| 1 (cluster-id) | The cluster ID of the branches which will be listed | Yes      | Only works in non-interactive mode.   |
```
Comment (zhangyangyu) [inline tidb-cloud/ticloud-branch-list.md:46]:
```suggestion
| <cluster-id> | The cluster ID of the branches which will be listed | Yes      | Only works in non-interactive mode.   |
```

File: tidb-cloud/ticloud-connect.md [hunk 4]
```diff
@@ -49,6 +52,7 @@ In non-interactive mode, you need to manually enter the required flags. In inter
 | Flag                    | Description                       | Required | Note                                                 |
 |-------------------------|-----------------------------------|----------|------------------------------------------------------|
 | -c, --cluster-id string | Cluster ID                        | Yes      | Only works in non-interactive mode.                  |
+| -c, --branch-id string  | Branch ID                         | No       | Only works in non-interactive mode.                  |
```
Comment (zhangyangyu) [inline tidb-cloud/ticloud-connect.md:55]:
```suggestion
| -b, --branch-id string  | Branch ID                         | No       | Only works in non-interactive mode.                  |
```

File: tidb-cloud/ticloud-branch-create.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
+---
+title: ticloud branch create
+summary: The reference of `ticloud branch create`.
+---
+
+# ticloud branch create
+
+Create a branch:
```
Comment (qiancai) [inline tidb-cloud/ticloud-branch-create.md:8]:
```suggestion
Create a branch for a cluster:
```

File: tidb-cloud/ticloud-branch-create.md (hunk 1 above)
Comment (qiancai) [inline tidb-cloud/ticloud-branch-create.md:16]:
```suggestion
> Currently, you can only create branches for [TiDB Serverless](/tidb-cloud/select-cluster-tier.md#tidb-serverless-beta) clusters.
```

File: tidb-cloud/ticloud-branch-create.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+```
+
+## Flags
+
+In non-interactive mode, you need to manually enter the required flags. In interactive mode, you can just follow CLI prompts to fill them in.
+
+| Flag                    | Description                                                | Required | Note                                                |
+|-------------------------|------------------------------------------------------------|----------|-----------------------------------------------------|
+| -c, --cluster-id string | The ID of the cluster, in which the branch will be created | Yes      | Only works in non-interactive mode.                 |
+| --branch-name string    | Name of the branch to be created                           | Yes      | Only works in non-interactive mode.                 |
```
Comment (qiancai) [inline tidb-cloud/ticloud-branch-create.md:39]:
```suggestion
| --branch-name string    | The name of the branch to be created                           | Yes      | Only works in non-interactive mode.                 |
```

File: tidb-cloud/ticloud-branch-describe.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
+---
+title: ticloud branch describe
+summary: The reference of `ticloud branch describe`.
+---
+
+# ticloud branch describe
+
+Get information about a branch (such as the endpoints, user_prefix, and usages):
```
Comment (qiancai) [inline tidb-cloud/ticloud-branch-describe.md:8]:
```suggestion
Get information about a branch (such as the endpoints, [user name prefix](/tidb-cloud/select-cluster-tier.md#user-name-prefix), and usages):
```

File: tidb-cloud/ticloud-branch-describe.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
```
Comment (qiancai) [inline tidb-cloud/ticloud-branch-describe.md:9]: "and usages" 里的 "usages" 是什么

File: tidb-cloud/ticloud-branch-list.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
+---
+title: ticloud branch list
+summary: The reference of `ticloud branch list`.
+---
+
+# ticloud branch list
+
+List all branches in a cluster:
```
Comment (qiancai) [inline tidb-cloud/ticloud-branch-list.md:8]:
```suggestion
List all branches for a cluster:
```

File: tidb-cloud/ticloud-branch-list.md [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+
+Or use the following alias command:
+
+```shell
+ticloud branch ls <cluster-id> [flags]
+```
+
+## Examples
+
+List all branches in a cluster (interactive mode):
```
Comment (qiancai) [inline tidb-cloud/ticloud-branch-list.md:22]:
```suggestion
List all branches for a cluster (interactive mode):
```

File: tidb-cloud/ticloud-branch-list.md [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+
+```shell
+ticloud branch list
+```
+
+List all branches in a specified cluster (non-interactive mode):
```
Comment (qiancai) [inline tidb-cloud/ticloud-branch-list.md:28]:
```suggestion
List all branches for a specified cluster (non-interactive mode):
```

File: tidb-cloud/ticloud-branch-list.md [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+
+```shell
+ticloud branch list <cluster-id>
+```
+
+List all branches in a specified cluster in the JSON format:
```
Comment (qiancai) [inline tidb-cloud/ticloud-branch-list.md:34]:
```suggestion
List all branches for a specified cluster in the JSON format:
```

File: tidb-cloud/ticloud-branch-list.md [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+ticloud branch list <cluster-id> -o json
+```
+
+## Args
+
+Branch list command has the following arguments:
+
+| Argument Index | Description                                         | Required | Note                                  |
+|----------------|-----------------------------------------------------|----------|---------------------------------------|
+| <cluster-id> | The cluster ID of the branches which will be listed | Yes      | Only works in non-interactive mode.   |
```
Comment (qiancai) [inline tidb-cloud/ticloud-branch-list.md:46]: 这条可以是 Flag 吗

File: tidb-cloud/ticloud-branch-describe.md (hunk 8 above)
Comment (shiyuhang0) [reply tidb-cloud/ticloud-branch-describe.md:9]: 计费里的东西，包括 ru，storage 这些。大概有这些：

File: tidb-cloud/ticloud-branch-list.md (hunk 13 above)
Comment (shiyuhang0) [reply tidb-cloud/ticloud-branch-list.md:46]: 我本来也想设计成 flag，但 cluster 那边设计成了 arg. 为了保持一致我就也这样设计了

File: tidb-cloud/ticloud-branch-create.md [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+
+Create a branch for a cluster:
+
+```shell
+ticloud branch create [flags]
+```
+
+> **Note:**
+>
+> Currently, you can only create branches for TiDB Serverless cluster.
```
Comment (ran-huang) [inline tidb-cloud/ticloud-branch-create.md:16]:
```suggestion
> Currently, you can only create branches for [TiDB Serverless](/tidb-cloud/select-cluster-tier.md#tidb-serverless) cluster.
```

File: tidb-cloud/ticloud-branch-list.md [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+ticloud branch list <cluster-id>
+```
+
+List all branches for a specified cluster in the JSON format:
+
+```shell
+ticloud branch list <cluster-id> -o json
+```
+
+## Args
```
Comment (ran-huang) [inline tidb-cloud/ticloud-branch-list.md:40]:
```suggestion
## Arguments
```

File: tidb-cloud/ticloud-branch-list.md [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+
+Branch list command has the following arguments:
```
Comment (ran-huang) [inline tidb-cloud/ticloud-branch-list.md:42]:
```suggestion
The `branch list` command has the following arguments:
```

**Review by qiancai — APPROVED** [review]
/lgtm

**Review by zhangyangyu — APPROVED** [review]

**Pull request merged**